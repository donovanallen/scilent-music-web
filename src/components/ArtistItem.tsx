import Link from 'next/link';
import React from 'react';
import { FaUser } from 'react-icons/fa6';

import NextImage from '@/components/NextImage';
import Pill from '@/components/Pill';

interface ArtistItemProps {
  id?: string;
  name?: string;
  image?: string;
  timestamp?: string | Date;
  type?: string;
}

const ArtistItem: React.FC<ArtistItemProps> = ({ id, name, image, type }) => {
  const imagePath = image;

  return (
    <Link href={`/artist/${id}`} aria-label={`Go to artist ${name}`}>
      <div className='rounded-md overflow-hidden bg-neutral-900 hover:text-brand-light hover:border-brand-light hover:border transition'>
        <div className='relative aspect-square rounded-md w-full bg-neutral-700 animated-underline'>
          {imagePath ? (
            <NextImage
              src={imagePath}
              alt={`Image of artist: ${name}`}
              fill
              className='aspect-square object-cover hover:pb-1 transition'
              useSkeleton
            />
          ) : (
            <FaUser size={64} className='m-auto h-full text-dark' />
          )}
          {type && (
            <Pill
              text={type}
              className='absolute bottom-2 right-2 bg-neutral-800/90'
            />
          )}
        </div>
        <div className='relative p-3 w-full overflow-clip'>
          <h4 className='truncate'>{name}</h4>
        </div>
      </div>
    </Link>
  );
};

export default ArtistItem;
