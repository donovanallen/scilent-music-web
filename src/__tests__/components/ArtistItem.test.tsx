import { render } from '@testing-library/react';
import React from 'react';

import ArtistItem from '@/components/ArtistItem';

describe('ArtistItem', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<ArtistItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
