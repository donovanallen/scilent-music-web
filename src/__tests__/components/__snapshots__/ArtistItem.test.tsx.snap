// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistItem renders correctly 1`] = `
<DocumentFragment>
  <a
    aria-label="Go to artist undefined"
    href="/artist/undefined"
  >
    <div
      class="rounded-md overflow-hidden bg-neutral-900 hover:text-brand-light hover:border-brand-light hover:border transition"
    >
      <div
        class="relative aspect-square rounded-md w-full bg-neutral-700 animated-underline"
      >
        <svg
          class="m-auto h-full text-dark"
          fill="currentColor"
          height="64"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
          />
        </svg>
      </div>
      <div
        class="relative p-3 w-full overflow-clip"
      >
        <h4
          class="truncate"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;
